Apps/Gate/gate.c

```diff
@@ -43,57 +48,34 @@ void Gate_Init(void)
  ******************************************************************************/
 void Gate_Loop(void)
 {
-#ifndef GATE_POLLING
-    static uint8_t first_conversion = 1;
-#endif
-    static short pipe_id = 0;
-#ifndef NODETECTION
-    static short previous_id = -1;
-#endif
-    static volatile bool gate_running = false;
-    static uint32_t last_time         = 0;
+    static short pipe_id      = 0;
+    static uint32_t last_time = 0;
 
     // Check the detection status.
-    if (RoutingTB_IDFromService(gate) == 0)
-    {
-#ifndef NODETECTION
-        // We don't have any ID, meaning no detection occure or detection is occuring.
-        if (previous_id == -1)
-        {
-            // This is the start period, we have to make a detection.
-            // Be sure the network is powered up 20 ms before starting a detection
-            if (Luos_GetSystick() > 20)
-            {
-                // No detection occure, do it
-                RoutingTB_DetectServices(gate);
-                previous_id = 0;
-#ifndef GATE_POLLING
-                first_conversion = 1;
-                update_time      = TimeOD_TimeFrom_s(GATE_REFRESH_TIME_S);
-#endif
-            }
-        }
-#endif
-        pipe_id = 0;
-    }
-    else
+    if (Luos_IsNodeDetected())
     {
         // Network have been detected, We are good to go
         if (pipe_id == 0)
         {
             // We dont have spotted any pipe yet. Try to find one
             pipe_id = PipeLink_Find(gate);
+            if (pipe_id > 0)
+            {
+                gate_running = true;
+                return;
+            }
         }
         if (gate_running && !detection_ask)
         {
             // Manage input and output data
             if ((Luos_GetSystick() - last_time >= TimeOD_TimeTo_ms(update_time)) && (Luos_GetSystick() > last_time))
             {
                 last_time = Luos_GetSystick();
-                DataManager_Run(gate);
-#ifndef GATE_POLLING
-                if (first_conversion == 1)
+                if (end_detection)
                 {
```

Comment:
I don't understand why you created a dedicated variable instead of using Luos_IsNodeDetected() function

Reply:
because this variable shows that a new detection has happened so we can send the auto updates to the services.

Reply:
Ok I get it.